### PR TITLE
Build warnings cleanup

### PR DESCRIPTION
**Pull request by deltaoscarmike**
- Small memory leak fixed
- Code cleanup to silent build warnings
- Removing of missing-declarations warning because it requires too many fixes and is not critical

### CODE REVIEW

File: src/projectM-sdl/pmSDL.cpp [hunk 1]
```diff
@@ -30,12 +30,12 @@ void projectMSDL::audioInputCallbackS16(void *userdata, unsigned char *stream, i
     app->pcm()->addPCM16(pcm16);
 }
 
-SDL_AudioDeviceID projectMSDL::selectAudioInput(int count) {
+SDL_AudioDeviceID projectMSDL::selectAudioInput(int _count) {
     // ask the user which capture device to use
     // printf("Please select which audio input to use:\n");
     printf("Detected devices:\n");
-    for (int i = 0; i < count; i++) {
-        printf("  %i: 🎤%s\n", i, SDL_GetAudioDeviceName(i, true));
```
Comment (revmischa) [inline src/projectM-sdl/pmSDL.cpp]: :(

File: src/libprojectM/MilkdropPresetFactory/InitCond.cpp [hunk 2]
```diff
@@ -138,7 +138,7 @@ void InitCond::init_cond_to_string() {
 		return;
 	
 	/* Copy the string into the initial condition string buffer */	
-	strncpy(init_cond_string_buffer + init_cond_string_buffer_index, string, string_length);
+    memcpy(init_cond_string_buffer + init_cond_string_buffer_index, string, string_length + 1);
```
[the remark this answers is not in the document]
Comment (deltaoscarmike) [reply src/libprojectM/MilkdropPresetFactory/InitCond.cpp]: It's an attempt to resolve a gcc 8 warning:
`output truncated before terminating nul copying as many bytes from a string as its length`

File: src/projectM-sdl/pmSDL.cpp (hunk 1 above)
Comment (deltaoscarmike) [reply src/projectM-sdl/pmSDL.cpp]: Why the special character in printf before %s

**Comment by deltaoscarmike** [comment]
It will fix #114

File: src/projectM-sdl/pmSDL.cpp (hunk 1 above)
Comment (revmischa) [reply src/projectM-sdl/pmSDL.cpp]: its just an emoji representing an audio input device. adds flair. not important 👨‍🍳

**Pull request merged**